### PR TITLE
[Alpha][13.0] Add stock_cubiscan

### PR DESCRIPTION
**Pull request by guewen**
[Cubiscan](https://cubiscan.com/) are dimensioners for cubing and weighing in
warehouses.  This module implements the communication with the dimensioners as
well as a screen to measure and weight packaging of the products.

### CODE REVIEW

File: stock_cubiscan/models/stock.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    _inherit = "stock.warehouse"
+
+    cubiscan_device_ids = fields.One2many(
+        "cubiscan.device", "warehouse_id", string="Cubiscan Devices"
+    )
+
+
+class ProductPackaging(models.Model):
+    _inherit = "product.packaging"
+    # FIXME: move this in product_packaging_type
```
Comment (simahawk) [inline stock_cubiscan/models/stock.py]: @guewen still the case?

File: stock_cubiscan/wizard/cubiscan_wizard.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+
+
+class CubiscanWizard(models.TransientModel):
```
Comment (simahawk) [inline stock_cubiscan/wizard/cubiscan_wizard.py:7]: a docstring explaining what it does?

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by simahawk** [comment]
@guewen rebase/pre-commit needed.
@p-tombez pls pass by for a review ;)

File: stock_cubiscan/models/stock.py (hunk 1 above)
Comment (guewen) [reply stock_cubiscan/models/stock.py]: yes, it should be in this module IMO https://github.com/OCA/product-attribute/pull/536

**Review by p-tombez — COMMENTED** [review]
LGTM, just need to fix the dependencies to make Travis happy

**Review by jgrandguillaume — APPROVED** [review]
Functional review ok on my side

**Comment by guewen** [comment]
@p-tombez the dependency (https://github.com/OCA/web/pull/1480) has been merged, I amended the last commit to trigger travis and runbot builds

**Comment by simahawk** [comment]
tests are :red_circle:

**Comment by guewen** [comment]
@simahawk fixed

File: stock_cubiscan/models/cubiscan.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    @api.constrains("device_address", "port")
+    def _check_connection_infos(self):
+        self.ensure_one()
+        if not 1 <= self.port <= 65535:
+            raise ValidationError(_("Port must be in range 1-65535"))
+
+    def copy(self, default=None):
+        if not default:
+            default = {}
+        default["state"] = "not_ready"
```
Comment (simahawk) [inline stock_cubiscan/models/cubiscan.py]: wondering: couldn't we achieve the same by setting `copy=False` on the field?

File: stock_cubiscan/models/stock.py (hunk 1 above)
Comment (simahawk) [reply stock_cubiscan/models/stock.py]: can you update the comment w/ a link to https://github.com/OCA/product-attribute/tree/13.0/product_packaging_type ?

**Review by simahawk — APPROVED** [review]
@guewen 2 little questions. LG anyway.

**Comment by guewen** [comment]
@simahawk yes, you are right, I amended the migration commit with your 2 remarks

**Pull request merged**